src/core/client-app.ts

```diff
@@ -139,6 +142,19 @@ export class ClientApp
         return this;
     }
 
+    public registerDialogService(dialogService: DialogService): this
+    {
+        given(dialogService, "dialogService").ensureHasValue().ensureIsObject();
+
+        if (this._isBootstrapped)
```

Comment:
this is an outdated pattern. Do the following:
given(this, "this").ensure(t => !t._isBootstrapped, "already bootstrapped");